pylint/checkers/base/function_checker.py

```diff
@@ -0,0 +1,222 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Function checker for Python code."""
+
+from __future__ import annotations
+
+from itertools import chain
+
+from astroid import nodes
+
+from pylint.checkers import utils
+from pylint.checkers.base.basic_checker import _BasicChecker
+from pylint.lint.pylinter import PyLinter
+
+
+class FunctionChecker(_BasicChecker):
+    """Check if a function definition handles possible side effects."""
+
+    msgs = {
+        "W0135": (
+            "The code after line %s will never be executed ('GeneratorExit' needs to be handled)",
```

Comment:
Yah this was a pretty rough guess at what to use based on the original direction. Since we raise on contextmanager decorated function, should this instead talk about the context might not be properly cleaned up based on usage?

Reply:
Sounds good, happy to review the text when you give it a go.

Reply:
@jacobtylerwalls edited the message to be more explicity about the contextmanager might not exit cleanly and how to handle